bundles/org.openhab.binding.vigicrues/src/main/java/org/openhab/binding/vigicrues/internal/handler/VigiCruesHandler.java

```diff
@@ -102,26 +211,42 @@ public void handleCommand(ChannelUID channelUID, Command command) {
-                String response = HttpUtil.executeUrl("GET", queryUrl, TIMEOUT_MS);
-                updateStatus(ThingStatus.ONLINE);
-                OpenDatasoftResponse apiResponse = gson.fromJson(response, OpenDatasoftResponse.class);
-                Arrays.stream(apiResponse.getRecords()).findFirst().flatMap(Record::getFields).ifPresent(field -> {
-                    field.getHeight().ifPresent(height -> updateQuantity(HEIGHT, height, METRE));
-                    field.getFlow().ifPresent(flow -> updateQuantity(FLOW, flow, CUBICMETRE_PER_SECOND));
-                    field.getTimestamp().ifPresent(date -> updateDate(OBSERVATION_TIME, date));
+            OpenDatasoftResponse measures = apiHandler.getMeasures(config.id);
+            measures.getFirstRecord().ifPresent(field -> {
+                field.getHeight().ifPresent(height -> {
+                    updateQuantity(HEIGHT, height, SIUnits.METRE);
+                    updateRelativeMeasure(RELATIVE_HEIGHT, referenceHeights, height);
                 });
-            } else {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.DISABLED,
-                        "queryUrl should never be null, but it is !");
+                field.getFlow().ifPresent(flow -> {
+                    updateQuantity(FLOW, flow, SmartHomeUnits.CUBICMETRE_PER_SECOND);
+                    updateRelativeMeasure(RELATIVE_FLOW, referenceFlows, flow);
+                });
+                field.getTimestamp().ifPresent(date -> updateDate(OBSERVATION_TIME, date));
+            });
+            if (portion != null) {
+                InfoVigiCru status = apiHandler.getTronconStatus(portion);
+                updateAlert(ALERT, status.vicInfoVigiCru.vicNivInfoVigiCru - 1);
+                updateString(SHORT_COMMENT, status.vicInfoVigiCru.vicSituActuInfoVigiCru);
+                updateString(COMMENT, status.vicInfoVigiCru.vicQualifInfoVigiCru);
             }
-        } catch (MalformedURLException e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    String.format("Querying '%s' raised : %s", queryUrl, e.getMessage()));
-        } catch (IOException e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                    String.format("Error opening connection to VigiCrues webservice : {}", e.getMessage()));
+            updateStatus(ThingStatus.ONLINE);
+        } catch (VigiCruesException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.DISABLED, e.getMessage());
+        }
+    }
+
+    private void updateString(String channelId, String value) {
+        if (isLinked(channelId)) {
+            updateState(channelId, new StringType(value));
+        }
+    }
+
+    private void updateRelativeMeasure(String channelId, List<QuantityType<?>> reference, double value) {
+        if (reference.size() > 0) {
+            double percent = value / reference.get(0).doubleValue() * 100;
```

Comment:
What is the point of keeping track of multiple reference values if you only care about the first one?

Reply:
I currently use the lowest value as reference, but maybe we could introduce a setting to let the user decide if he wants it compared to lowest or highest. Reason why I did it this way.